src/batchview.jl

```diff
@@ -144,6 +139,15 @@ function _getbatch(A::BatchView{TElem, TData, Val{nothing}}, obsindices) where {
     getobs(A.data, obsindices)
 end
 
+function getobs!(buffer, A::BatchView{TElem, TData, Val{nothing}}, i) where {TElem, TData}
+    obsindices = _batchindexes(A, i)
+    return _getbatch!(buffer, A, obsindices)
+end
+
+function _getbatch!(buffer, A::BatchView{TElem, TData, Val{nothing}}, obsindices) where {TElem, TData}
+    return getobs!(buffer, A.data, obsindices)
+end
```

Comment:
I would've assumed this also works for `collate=true|false`, because unlike `getobs`/`_getbatch` the caller has already figured out the return type by passing in `buffer`. Not sure I understand the interface or API contract that well though, so please correct me if not.